### PR TITLE
Setup the NotificationsViewController and the RequestTableViewCell

### PR DESCRIPTION
**Pull request by tiffany-pann**
## Overview
- Created the NotificationsViewController and the RequestTableViewCell
- Added a responseRequest model

## Changes Made
- Set up the notificationVC. The user can tap on the notification button in the home screen to push this new VC. 
- Created the table view and its respective table view cell. Initialized the cell in two different ways, depending on whether or not the request is pending or if the user has already accepted/denied it. 
- Created a tentative responseRequest model that may need to be updated depending on the backend.

## Test Coverage
- Manual testing with making sure that the tableview looks ok, button functionality still not yet implemented. 

## Next Steps (delete if not applicable)
- Replace currently hard-coded data with networking calls 
- Add functionality to the buttons ^ in relation to network calls and updating the cell when the user has clicked on either accept/deny

## Screenshots (delete if not applicable)
image
image

### CODE REVIEW

File: Scoop/Controller/Home/HomeViewController.swift [hunk 1]
```diff
@@ -125,6 +127,26 @@ class HomeViewController: UIViewController {
         signOutButton.addAction(signOutAction, for: .touchUpInside)
     }
     
+    private func setupNotificationButton() {
+        notificationButton.setImage(UIImage(named: "notification"), for: .normal)
+        notificationButton.translatesAutoresizingMaskIntoConstraints =  false
```
Comment (ReadeP) [inline Scoop/Controller/Home/HomeViewController.swift]: you don't need this line with snapkit

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        self.navigationItem.setHidesBackButton(true, animated: true)
+        
+        setupBackButton()
+        setupTableView()
+    }
+    
+    private func setupBackButton() {
+        backButton.setImage(UIImage(named: "BackArrow"), for: .normal)
+        backButton.addTarget(self, action: #selector(popVC), for: .touchUpInside)
+        self.navigationController?.navigationBar.addSubview(backButton)
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: 1. remove `self`
2. Add newline between addSubview and setting constraints

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            make.leading.equalToSuperview().inset(25)
+            make.centerY.equalToSuperview()
+        }
+    }
+
+    private func setupTableView() {
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.backgroundColor = .white
+        tableView.separatorInset = UIEdgeInsets.zero
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: `.zero`

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        tableView.contentInset = UIEdgeInsets(top: 20, left: 0, bottom: 20, right: 0)
+        tableView.separatorColor = .systemGray
+        tableView.separatorStyle = .singleLine
+        // Not sure about this, it makes the entire cell layout shift. Not sure how else to make the separator last the entire line though. 
+//        tableView.layoutMargins = UIEdgeInsets.zero
+        tableView.register(RequestTableViewCell.self, forCellReuseIdentifier: requestCellIdentifier)
+        view.addSubview(tableView)
+        
+        tableView.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(-35)
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: `inset(35)`

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            make.leading.trailing.equalToSuperview()
+            make.bottom.equalTo(view.safeAreaLayoutGuide)
+        }
+    }
+    
+    @objc func popVC() {
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: make `private`

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        self.navigationController?.popViewController(animated: true)
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: remove `self`

File: Scoop/Controller/Home/NotificationsViewController.swift (hunk 5 above)
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: make `private`

File: Scoop/Controller/Home/NotificationsViewController.swift [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+// MARK: UITableViewDataSource
+extension NotificationsViewController: UITableViewDataSource {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        requests.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: requestCellIdentifier) as! RequestTableViewCell
+        if requests[indexPath.row].approved {
```
Comment (ReadeP) [inline Scoop/Controller/Home/NotificationsViewController.swift]: one line `cell.configure(request: requests[indexPath.row], status: requests[indexPath.row].approved)`

File: Scoop/Controller/Home/RequestTableViewCell.swift [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+        } else {
+            // Fallback on earlier versions
+            acceptButton.setTitle("Accept", for: .normal)
+            acceptButton.setTitleColor(.white, for: .normal)
+            acceptButton.layer.cornerRadius = 10
+            acceptButton.backgroundColor = UIColor.scoopGreen
+        }
+        
+        acceptButton.addTarget(self, action: #selector(acceptRequest), for: .touchUpInside)
+        contentView.addSubview(acceptButton)
```
Comment (ReadeP) [inline Scoop/Controller/Home/RequestTableViewCell.swift:119]: newline

File: Scoop/Model/Ride.swift [hunk 9]
```diff
@@ -22,6 +22,14 @@ struct Ride: Codable {
     var type: String
 }
 
+/// This version of the Ride model is used in the approving/denying rides networking request. Is there a better way to do this? (only wanting to include certain fields of a model)
+struct truncRide: Codable {
```
Comment (ReadeP) [inline Scoop/Model/Ride.swift]: `TruncRide`

**Review by ReadeP — APPROVED** [review]
🔥🔥🔥

File: Scoop/Controller/Home/RequestTableViewCell.swift [hunk 10]
```diff
@@ -0,0 +1,168 @@
+//
+//  ReuqestTableViewCell.swift
```
Comment (vinnie4k) [inline Scoop/Controller/Home/RequestTableViewCell.swift:2]: "Reuqest" lol

**Review by vinnie4k — APPROVED** [review]
niceee

**Review by rs929 — APPROVED** [review]
Nice! LGTM. 👍

File: Scoop/Controller/Home/RequestTableViewCell.swift (hunk 10 above)
Comment (tiffany-pann) [reply Scoop/Controller/Home/RequestTableViewCell.swift:2]: spelling is hard :[

**Pull request merged**